streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -633,7 +633,7 @@ public boolean test(final K1 key, final V1 value) {
             .withKeys(keySerde)
             .withValues(valueSerde)
             .persistent()
-            .windowed(windows.maintainMs(), windows.segments, true)
+            .windowed(windows.after, windows.maintainMs(), windows.segments, true)
```

Comment:
Given Damian's and Matthias' discussion about about `JoinWindows#size()`, it's not just a matter about consistency but also (and more importantly) a matter of correctness?